User tools support over HTTPS in console (#704)
neo4j/docs-aura: modules/ROOT/pages/security/secure-connections.adoc

To continue accessing Browser and Bloom, you can configure a VPN in your VPC and connect to these services over the VPN.

> == Explore and Query access

Reviewer: ```suggestion
== Tool access
```

> When public traffic is disabled, Query and Browser are not accessible via the public internet. 

Reviewer: ```suggestion
When public traffic is disabled, Query and Bloom are not accessible via the public internet. 
```

> To continue accessing these tools, link:https://neo4j.com/docs/aura/getting-started/connect-instance/#_connection_method[connect via HTTPS (port 443)], this is helpful when network security blocks Bolt (port 7687), e.g. when a private link is setup on the database with public traffic disabled.

Reviewer: The link should be an xref.

Author: Fixed it

So if you've set one up for `us-east-1` then those network connections will apply to all instances in that region.
You can set up a second private link connection to applications that are hosted in a second region (for example `us-west-1`) but still housed inside the same Aura project.

[.tabbed-example]
====
[.include-with-AWS-using-PrivateLink]
=====

> For a step-by-step guide, see the link:https://neo4j.com/blog/auradb/neo4j-aws-privatelink-configuration/#2[How to Configure Neo4j Aura With AWS PrivateLink] blog article.

Reviewer: Not a huge fan of linking to blog posts, do we really need it?

Author: Yes! We really do need it

> Refer to https://aws.amazon.com/privatelink[AWS PrivateLink] docs for IAM requirements.

Reviewer: ```suggestion
Refer to link:https://aws.amazon.com/privatelink[AWS PrivateLink] docs for IAM requirements.
```

AWS region `us-east-1` does not support AZ `use1-az3` for private endpoints.

image::privatelink.png["VPC connectivity with AWS PrivateLink"]

Without private endpoints, you access Query and Explore over the internet:

image::privatelink_01_before_enabling.png["Architecture overview before enabling private endpoints"]

> When you have enabled private endpoints and disabled public internet access, you can no longer connect Browser or Bloom to your instances over the internet and therefore to continue accessing Query and Explore, you can either connect via HTTPS (port 443), or use a private endpoint.

Reviewer: ```suggestion
When you have enabled private endpoints and disabled public internet access, you can no longer connect Query and Explore to your instances over the internet.
To continue accessing the tools, you can either connect via HTTPS (port 443), or use a private endpoint.
```

To access Query and Explore over a VPN, you must ensure that:

* The VPN server uses the https://docs.aws.amazon.com/vpc/latest/userguide/vpc-dns.html#AmazonDNS[VPC's DNS servers].
* You use the *Private URI* shown on the instance tile and in the instance details.

> It will be different from the *Connection URI* you used before.

Reviewer: ```suggestion
It is different from the *Connection URI* you used before.
```

image::privatelink_03_browser_bloom_over_vpn.png["Accessing Query and Explore over a VPN"]
=====

[.include-with-GCP-using-Private-Service-Connect]
=====

Refer to https://cloud.google.com/vpc/docs/private-service-connect[GCP Private Service Connect] docs for required permissions.

image::privateserviceconnect.png["VPC connectivity with GCP Private Service Connect"]

Without private endpoints, you access Query and Explore over the internet:

image::privateserviceconnect_01_before_enabling.png["Architecture overview before enabling private endpoints"]

> When you have enabled private endpoints and disabled public internet access, you can no longer connect Browser or Bloom to your instances over the internet and therefore to continue accessing Query and Explore, you can either connect via HTTPS (port 443), or use a private endpoint.

Reviewer: As above

Author: fixed!

To access Query and Explore over a VPN, you must ensure that:

* You have set up link:https://cloud.google.com/dns/docs/zones/manage-response-policies[GCP Response Policy Zone], or an equivalent DNS service, inside of the VPC.
* You use the *Private URI* shown on the instance tile and in the instance details.

> It will be different from the *Connection URI* you used before.

Reviewer: ```suggestion
It is different from the *Connection URI* you used before.
```

image::privateserviceconnect_03_browser_bloom_over_vpn.png["Accessing Query (Browser) and Explore (Bloom) over a VPN"]
=====

[.include-with-Azure-using-Private-Link]
=====

> Refer to https://azure.microsoft.com/en-us/products/private-link/#overview[Azure Private Link] docs to create an endpoint in your Virtual Network (VNet) that connects to Aura.

Reviewer: ```suggestion
Refer to link:https://azure.microsoft.com/en-us/products/private-link/#overview[Azure Private Link] docs to create an endpoint in your Virtual Network (VNet) that connects to Aura.
```

image::azure_privatelink.png["VNet connectivity with Azure Private Link"]

Without private endpoints, you access Query and Explore over the internet:

image::azure_privatelink_01_before_enabling.png["Architecture overview before enabling private endpoints"]

> When you have enabled private endpoints and disabled public internet access, you can no longer connect Browser or Bloom to your instances over the internet and therefore to continue accessing Query and Explore, you can either connect via HTTPS (port 443), or use a private endpoint.

Reviewer: ```suggestion
When you have enabled private endpoints and disabled public internet access, you can no longer connect Query or Explore to your instances over the internet.
To continue accessing the tools, you can either connect via HTTPS (port 443), or use a private endpoint.
```

To access Query and Explore over a VPN, you must ensure that:

* You have set up link:https://cloud.google.com/dns/docs/zones/manage-response-policies[GCP Response Policy Zone], or an equivalent DNS service, inside of the VPC.
* You use the *Private URI* shown on the instance tile and in the instance details.

> It will be different from the *Connection URI* you used before.

Reviewer: ```suggestion
It is different from the *Connection URI* you used before.
```
